semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -471,30 +471,17 @@ let check_function_signature env fun_exp args_taints =
   in
   match (!hook_function_taint_signature, fun_exp) with
   | ( Some hook,
+      (* Case `$F()` *)
       {
-        e =
-          Fetch
-            {
-              base =
-                Var
-                  {
-                    ident;
-                    id_info =
-                      {
-                        G.id_resolved =
-                          {
-                            contents =
-                              Some ((G.ImportedEntity _ | G.ResolvedName _), _);
-                          };
-                        _;
-                      };
-                    _;
-                  };
-              rev_offset = _;
-              _;
-            };
+        e = Fetch { base = Var { ident; _ }; rev_offset = []; _ };
         eorig = SameAs eorig;
         _;
+      } )
+  | ( Some hook,
+      (* Case `$X. ... .$F()` *)
+      {
+        e = Fetch { base = _; rev_offset = Dot { ident; _ } :: _; _ };
```

Comment:
Why do you need to distinguish those two cases?

Reply:
It's for `ident`, so that we can have the token of the function. I can clarify

Reply:
Actually, I'll just perform that split lower since we only need it in one case